packages/wdio-lambdatest-service/src/launcher.js

```diff
@@ -0,0 +1,124 @@
+            ...this.options.lambdatestOpts
+        }
+
+        this.lambdatestTunnelProcess = new LambdaTestTunnelLauncher()
+
+        if (Array.isArray(capabilities)) {
+            capabilities.forEach(capability => {
+                capability.tunnel = true
+            })
+        } else if (typeof capabilities === 'object') {
+            capabilities.tunnel = true
+        }
+        // measure LT boot time
+        const obs = new PerformanceObserver(list => {
+            const entry = list.getEntries()[0]
+            log.info(
+                `LambdaTest Tunnel successfully started after ${entry.duration}ms`
+            )
+        })
+        obs.observe({ entryTypes: ['measure'], buffered: false })
+
+        let timer
+        performance.mark('ltTunnelStart')
+        return Promise.race([
+            new Promise((resolve, reject) => {
+                this.lambdatestTunnelProcess.start(tunnelArguments, err => {
+                    if (err) {
+                        return reject(err)
+                    }
+                    this.lambdatestTunnelProcess.getTunnelName(tunnelName => {
+                        if (Array.isArray(capabilities)) {
+                            capabilities.forEach(capability => {
+                                capability.tunnelName = tunnelName
+                            })
+                        } else if (typeof capabilities === 'object') {
+                            capabilities.tunnelName = tunnelName
+                        }
+                        resolve()
+                    })
+                })
+            }),
+            new Promise((resolve, reject) => {
+                timer = setTimeout(function() {
+                    reject(
+                        'LambdaTest Tunnel failed to start within 60 seconds!'
+                    )
+                }, 60000)
+            })
+        ]).then(
+            function(result) {
+                clearTimeout(timer)
+                performance.mark('ltTunnelEnd')
+                performance.measure('bootTime', 'ltTunnelStart', 'ltTunnelEnd')
+                return Promise.resolve(result)
+            },
+            function(err) {
+                clearTimeout(timer)
+                return Promise.reject(err)
+            }
+        )
+    }
+
+    onComplete() {
+        if (
+            !this.lambdatestTunnelProcess ||
+            typeof this.lambdatestTunnelProcess.isRunning !== 'function' ||
+            !this.lambdatestTunnelProcess.isRunning()
+        ) {
+            return
+        }
+
+        let timer
+        return Promise.race([
+            new Promise((resolve, reject) => {
+                this.lambdatestTunnelProcess.stop(err => {
+                    if (err) {
+                        return reject(err)
+                    }
+                    resolve()
+                })
+            }),
+            new Promise((resolve, reject) => {
+                timer = setTimeout(
```

Comment:
how about just having
```suggestion
                timer = setTimeout(() => reject(new Error(...)), TUNNEL_STOP_TIMEOUT)
```

Reply:
Done